model/histogram/histogram.go

```diff
@@ -453,3 +453,13 @@ func (c *cumulativeBucketIterator) At() Bucket[uint64] {
 		Index:          c.currIdx - 1,
 	}
 }
+
+func (h *Histogram) ReduceResolution(targetSchema int32) *Histogram {
```

Comment:
This needs a doc comment (including that the targetSchema must be smaller than the current schema).